linode_metadata/metadata_client.py

```diff
@@ -300,8 +315,34 @@ def _api_call(
         url = self._prepare_url(endpoint)
         request_params = self._get_api_call_params(url, headers, method, body)
 
+        if self._debug:
+            if self._debug_file is None:
+                self._print_request_debug_info(request_params, sys.stderr)
+            else:
+                if os.path.exists(self._debug_file):
+                    with open(self._debug_file, "a", encoding="UTF-8") as file:
+                        self._print_request_debug_info(request_params, file)
+                else:
+                    print(
+                        "No debug file exists to write. "
+                        + "Please check the file path or use default output."
```

Comment:
optional: It might make sense to raise this as an error rather than printing it, or alternatively it could be printed to stderr